repositories/employeeRepository.go

```diff
@@ -16,71 +16,317 @@ type EmployeeRepository struct {
-	commandTag, err := repository.DB.Exec(context.Background(),
-		"UPDATE employees SET firstname=$1, lastname=$2, idc=$3, c=$4, r=$5, u=$6, d=$7 WHERE id=$8;",
-		updatedEmp.FirstName, updatedEmp.LastName, updatedEmp.CompanyID, updatedEmp.C, updatedEmp.R, updatedEmp.U, updatedEmp.D, updatedEmp.ID)
+func (repository *EmployeeRepository) UpdateEmployee(employee models.Employee) error {
+	tx, err := repository.DB.Begin(context.Background())
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback(context.Background())
 
+	employeePers := persistence.Employees{
+		Firstname: employee.FirstName,
+		Lastname:  employee.LastName,
+		C:         employee.C,
+		R:         employee.R,
+		U:         employee.U,
+		D:         employee.D,
+	}
+	employeePers.Idc.Set(employee.CompanyID)
+	employeePers.Id.Set(employee.ID)
+
+	commandTag, err := employeePers.UpdateTx(&tx)
 	if err != nil {
-		fmt.Println(err)
 		return err
 	}
-	if commandTag.RowsAffected() != 1 {
+	if commandTag != 1 {
 		return errors.New("No row found to update")
 	}
+
+	tx.Commit(context.Background())
 	return nil
 }
 
 // DeleteEmployee .
 func (repository *EmployeeRepository) DeleteEmployee(id string) error {
-	commandTag, err := repository.DB.Exec(context.Background(), "DELETE FROM employees WHERE id=$1;", id)
+	tx, err := repository.DB.Begin(context.Background())
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback(context.Background())
 
+	employeePers := persistence.Employees{}
+	employeePers.Id.Set(id)
+
+	commandTag, err := employeePers.DeleteTx(&tx)
 	if err != nil {
 		return err
 	}
-	if commandTag.RowsAffected() != 1 {
+	if commandTag != 1 {
 		return errors.New("No row found to delete")
 	}
+
+	tx.Commit(context.Background())
 	return nil
 }
+
+// GetEmployeeExternalPermissions .
+func (repository *EmployeeRepository) GetEmployeeExternalPermissions(idReceivingCompany string, product models.Product) (models.ExternalRights, error) {
+	var allRights []models.ExternalRights
+	var rights models.ExternalRights
+
+	// 1. Using idReceivingCompany and idSharingCompany, acquire all external access rules for these two companies
+	tx, err := repository.DB.Begin(context.Background())
+	if err != nil {
+		return rights, err
+	}
+
+	defer tx.Rollback(context.Background())
+
+	queryExternalAccess := "SELECT * FROM external_access_rights WHERE idrc = $1 AND idsc = $2;"
+	rows, err := tx.Query(context.Background(), queryExternalAccess, idReceivingCompany, product.IDC)
+
+	if err != nil {
+		return rights, err
+	}
+	for rows.Next() {
+		var right models.ExternalRights
+		err := rows.Scan(&right.ID, &right.IDSC, &right.IDRC, &right.Read, &right.Update, &right.Delete, &right.Approved)
+		if err != nil {
+			return rights, err
+		}
+		allRights = append(allRights, right)
+	}
+	rows.Close()
+
+	if len(allRights) == 0 {
+		// 2a. If there is no rows returned employees from this company can't interact with this product
+		return rights, errors.New("You don't have any permission for this product")
```

Comment:
If you have an early return here, I guess there is no need for an: if - else if - else. 
I suggest maybe making a switch, and making the else as a default.